calc.rb

```diff
@@ -0,0 +1,76 @@
+puts "The Calculator"
+puts "I'll take two numbers and perform a mathematical operation on them. Let's try."
+
+# First number
+puts "\nWhat is the first number?"
+num1 = gets.chomp
+
+# Handling invalid user input for first number
+if num1 == nil || (num1.to_i.to_s != num1 && num1.to_f.to_s != num1)
```

Comment:
You've got `num1.to_i.to_s` here. You may accidentally truncate your input if it was a float. `gets.chomp` always gives us a string, so if the user typed in `1.5`, calling `.to_i` would truncate that to just `1`!